src/document.ts

```diff
@@ -8,14 +8,18 @@ import * as utils from './utils';
 const DEFAULT_RENDERER = 'njk';
 const DEFAULT_VIEW = '/views/base.njk';
 
+interface DocumentParts {
+  body?: string | null;
```

Comment:
I think you should either use the `?` or `| null` but not both (unless you have explicit usage for `undefined` vs `null`, but that doesn't seem to be the case).

Reply:
There are several checks that @jeremydw is doing that test against null across the codebase. If you take out the `| null` then `?:` is the same as doing `string | undefined`. Which doesn't allow for a null value.

Do we want up update the usage of `null` to `undefined` across more of the codebase to check for undefined instead of null?

Reply:
I'd appreciate some help in fixing this up if needed. :) Basically, the reason why the `body` can either be a `string` or `null` is because certain document types may or may not have bodies. For example, a Markdown doc obviously has a body (and potentially fields, if it has frontmatter). A YAML doc does not have a body.

Re-reading Steven's comment it looks like it's mostly a type declaration nit. I think this could be improved by changing it to:

`body: string | null`

That seems better to me, but the reason why `body` may be undefined (and I think why I had this as I did originally), is because the `body` *is* undefined all the way up until the doc's file is read and then set appropriately.

Reply:
I took out most of the `| null` usage that I had done, since it didn't really need a null value for most of what i was doing. For the `body` I made it `body?: string | null` and changed it so that the `undefined` body is representative of not having parsed the raw content. When the raw content is parsed the body either has a string value or is null if there is no body (such as with yaml files).

We could alternatively use an empty string to show that there is no content, but the content has been parsed. Thoughts?

Reply:
Just summarizing what Randy wrote out:

`body?: string | null` with the following:

- `undefined` meaning the raw content hasn't been parsed yet
- `null` meaning there's no body (i.e. YAML)
- `string` the actual raw content (i.e. for markdown files)

That makes perfect sense to me. 🙌  I think that's originally how I was thinking about it, as well. If that's what we want to go with, we can document it and it will appear in our docs for future contributors.